src/nvim/syntax.c

```diff
@@ -6859,6 +6857,7 @@ int hl_combine_attr(int char_attr, int prim_attr)
 
 /// \note this function does not apply exclusively to cterm attr contrary
 /// to what its name implies
+/// \warn don't call it with attr 0 (i.e., the null attribute)
```

Comment:
@teto was there a reason we can't assert() this?

Reply:
I think it's safe to assert on this, but it might be even better to remove the +1/-1 all around the code, which IIRC could be done in a backwards compatible way.